pkg/virt-controller/watch/clone/util.go

```diff
@@ -0,0 +1,92 @@
+package clone
+
+import (
+	"fmt"
+
+	"k8s.io/utils/pointer"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/api/snapshot/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	clonev1alpha1 "kubevirt.io/api/clone/v1alpha1"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+const (
+	vmKind           = "VirtualMachine"
+	kubevirtApiGroup = "kubevirt.io"
+)
+
+func getKey(name, namespace string) string {
+	return fmt.Sprintf("%s/%s", namespace, name)
+}
+
+func generateNameWithRandomSuffix(names ...string) string {
+	const randomStringLength = 5
+
+	if len(names) == 0 {
+		return ""
+	}
+
+	generatedName := names[0]
+	for _, name := range names[1:] {
+		generatedName = fmt.Sprintf("%s-%s", generatedName, name)
+	}
+
+	generatedName = fmt.Sprintf("%s-%s", generatedName, rand.String(randomStringLength))
+	return generatedName
+}
+
+func generateSnapshotName(cloneName, vmName string) string {
+	return generateNameWithRandomSuffix("clone", cloneName, "snapshot", vmName)
+}
+
+func generateRestoreName(cloneName, vmName string) string {
+	return generateNameWithRandomSuffix("clone", cloneName, "restore", vmName)
+}
+
+func generateVolumeName(volumeName string) string {
+	return generateNameWithRandomSuffix("clone", "volume", volumeName)
+}
+
+func generateVMName(oldVMName string) string {
+	return generateNameWithRandomSuffix(oldVMName, "clone")
+}
+
+func isInPhase(vmClone *clonev1alpha1.VirtualMachineClone, phase clonev1alpha1.VirtualMachineClonePhase) bool {
+	return vmClone.Status.Phase == phase
+}
+
+func generateSnapshot(vmClone *clonev1alpha1.VirtualMachineClone, sourceVM *v1.VirtualMachine) *v1alpha1.VirtualMachineSnapshot {
+	return &v1alpha1.VirtualMachineSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateSnapshotName(vmClone.Name, sourceVM.Name),
+			Namespace: sourceVM.Namespace,
+		},
```

Comment:
I think the snapshot and restore resources should be owned by clone resource.

Reply:
Thanks! That's an important note. Would take care of it. :+1:

Reply:
Done :ok_hand: